.github/workflows/run_detekt_and_unit_tests.yml

```
@@ -0,0 +1,38 @@
+name: Run Detekt and unit tests
+
+on: push
+
+jobs:
+  run_detekt_and_unit_tests:
+    name: Run Detekt and unit tests
+    runs-on: ubuntu-latest
+    timeout-minutes: 30
+    steps:
+      - name: Set up JDK 11
+        uses: actions/setup-java@v2
+        with:
+          distribution: 'temurin'
+          java-version: '11'
+
+      - name: Checkout source code
+        uses: actions/checkout@v2.3.2
+
+      - name: Cache Gradle
+        uses: actions/cache@v2
+        with:
+          path: |
+            ~/.gradle/caches/modules-*
+            ~/.gradle/caches/jars-*
+            ~/.gradle/caches/build-cache-*
+          key: ${{ runner.os }}-gradle-${{ hashFiles('**/*.gradle*') }}
+          restore-keys: |
+            ${{ runner.os }}-gradle-
+
+      - name: Run Detekt
+        run: ./gradlew detekt
+
+      - name: Run lint
+        run: ./gradlew lintStagingDebug
+
+      - name: Run unit tests with Kover
+        run: ./gradlew koverMergedHtmlReport
```

Could you please add this step too:

```suggestion
 - name: Archive code coverage reports
        uses: actions/upload-artifact@v2
        if: always()
        with:
          name: CodeCoverageReports
          path: build/reports/kover/merged/
```

So we will use the report from `kover` instead of `jacoco`